host-port.js

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var HostPort = {
+    validateHostPort: validateHostPort,
+    validateHost: validateHost,
+    validatePort: validatePort
+};
+
+module.exports = HostPort;
+
+function validateHost(host, allowEmphemeral) {
+    if (typeof host !== 'string') {
+        return 'Expected host to be a string';
+    }
+
+    var hostParts = host.split('.');
+    if (hostParts.length !== 4) {
+        return 'Expected host to contain IPv4';
+    }
+
+    var i;
+    for (i = 0; i < 4; i++) {
+        if (!stringIsValidNumber(hostParts[i])) {
+            return 'Expected each IP section to be integer';
```

Comment:
this error string is confusing, since an dotted-quad with non-numeric component isn't an IP address in the first place; calling it an "IP section" that's not an integer is confusing.

Please just use a simple regex, and say "Expected an IPv4 address" if not test